r/R/query-engine.R

```diff
@@ -20,11 +20,36 @@ do_exec_plan <- function(.data) {
   final_node <- plan$Build(.data)
   tab <- plan$Run(final_node)
 
+  # If arrange() created $temp_columns, make sure to omit them from the result
+  # We can't currently handle this in the ExecPlan itself because sorting
+  # happens in the end (SinkNode) so nothing comes after it.
   if (length(final_node$sort$temp_columns) > 0) {
-    # If arrange() created $temp_columns, make sure to omit them from the result
     tab <- tab[, setdiff(names(tab), final_node$sort$temp_columns), drop = FALSE]
   }
 
+  if (ncol(tab)) {
+    # Apply any column metadata from the original schema, where appropriate
+    original_schema <- source_data(.data)$schema
+    # TODO: do we care about other (non-R) metadata preservation?
```

Comment:
We already do discard other metadata when you convert from Arrow to R. Check out the example parquet file in inst/, for example: it has "pandas" metadata. I suspect it would be more surprising/alarming if we started messaging about this metadata all of the sudden.

Reply:
Hmm, Arrow to R conversion losing the metadata sounds like what I expected. TBH I _am_ a little surprised that the following happens silently, since the `tab` below _does_ have the pandas metadata there. Though that's been in Arrow for a while and if no one has complained, I guess it's not that important.

```
> tab <- read_parquet(system.file("v0.7.1.parquet", package = "arrow"), as_data_frame = FALSE)
> 
> dplyr::select(tab, carat)$metadata
NULL
```

Reply:
Actually, on 5.0:

```
new <- tab %>% 
  select(carat) %>% 
  compute()
new$metadata

$pandas
[1] "{\"index_columns\": [\"__index_level_0__\"], \"column_indexes\": [{\"name\":
...
```

I still don't think this is necessarily meaningful--especially when you're dealing with multiple files with potentially different metadata.